pkg/controller/cluster_controller.go

```diff
@@ -210,12 +216,19 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req reconcile.Request
 
 // SetupWithManager registers the controller with the controller.Manager from controller-runtime
 func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	var ingress client.Object
+	ingress = &v1.Ingress{}
+	if !util.CheckIfAPIVersionKindAvailable(mgr.GetConfig(), "networking.k8s.io/v1", "Ingress") {
```

Comment:
why do we need this - does it depend on the version of k8s being used or something?